tools/build_rules/verifier_test/kzip_archive.bzl

```diff
@@ -0,0 +1,122 @@
+# Copyright 2019 The Kythe Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Rule for creating generic .kzip compilation units using kzip create.
+"""
+
+def _key_pair(item):
+    return "{}={}".format(*item)
+
+def _kythe_uri(corpus, language, path = None, root = None):
+    uri = "kythe://{}?lang={}".format(corpus, language)
+    if path:
+        uri += "?path=" + path
+    if root:
+        uri += "?root=" + root
+    return uri
+
+def _single_path(srcs):
+    # If there is exactly one source, use that for the path.
+    # This is primarily to be compatible with some existing uses of kzip create.
+    if len(srcs) == 1:
+        return srcs[0].path
+    return None
+
+def _kzip_archive(ctx):
+    args = ctx.actions.args()
+    args.add("create")
+    args.add("--uri", _kythe_uri(
+        corpus = ctx.attr.corpus,
+        language = ctx.attr.lang,
+        path = _single_path(ctx.files.srcs),
+        root = ctx.attr.root,
+    ))
+    args.add("--output", ctx.outputs.kzip)
+    args.add("--output_key", str(ctx.label))
+    args.add_joined("--source_file", ctx.files.srcs, join_with = ",")
+    args.add_joined("--required_input", ctx.files.deps, join_with = ",")
```

Comment:
does required_input also need to include srcs?

Reply:
No, `kzip create` assumes that source files are also required input :-)